packages/SwingSet/src/kernel/kernel.js

```diff
@@ -733,14 +764,22 @@ export default function buildKernel(
 
     /** @type { PolicyInput } */
     const policyInput = harden(['create-vat', {}]);
+    // TODO: combine this with the return value from processStartVat
 
-    // eslint-disable-next-line no-use-before-define
-    return vatWarehouse
-      .createDynamicVat(vatID)
-      .then(makeSuccessResponse, makeErrorResponse)
-      .then(sendResponse)
-      .catch(err => console.error(`error in vat creation`, err))
-      .then(() => policyInput);
+    return (
+      // eslint-disable-next-line no-use-before-define
+      vatWarehouse
+        .createDynamicVat(vatID)
+        // if createDynamicVat fails, go directly to makeErrorResponse
+        .then(_ =>
+          enableSetup ? null : processStartVat({ type: 'startVat', vatID }),
```

Comment:
You know, it might make sense to send `dispatch.startVat` to `setup` based vats too, it would tell them when it's safe to make syscalls. I don't know how many unit tests would have to change, though, to add a no-op `startVat()` to their setup-based `dispatch` functions.

If we did that, this would become unconditional, and we wouldn't need to extract `enableSetup` first.

Reply:
We should document the conclusions we came to about the error handling of `processStartVat`. Something like:

If the new vat fails during module import or `buildRootObject`, the `startVat` delivery will signal a failure, and `deliverAndLogToVat` will `setTerminationTrigger`, which will unwind the crank after we return. `processStartVat` returns a PolicyInput rather than throwing, which means we'll call `makeSuccessResponse` (queueing a message to vatAdmin that the new vat was successfully created), but that message will be discarded when the crank is unwound.

Reply:
I was actually handling `startVat` in setup vats in the prior, abandoned (`buildRootObject` only) PR.  A lot of them already had code to ignore `bringOutYourDead` deliveries so adding another case to ignore was no big deal.  However, I'm not sure what benefit would be provided, as there's really no useful "you can now use syscalls" signal, because `setup` is called outside the crank by definition.  In a setup vat, you can use syscalls from calls to `dispatch` and you can't use them during `setup` or during module initialization.